File: 5/5.pas
program Problem5;

function GCD(a, b: Int64): Int64;
var
    t: Int64;
begin
    if a < 0 then a := -a;
    if b < 0 then b := -b;
    while b <> 0 do
    begin
        t := a mod b;
        a := b;
        b := t;
    end;
    GCD := a;
end;

function LCM(a, b: Int64): Int64;
begin
    if (a = 0) or (b = 0) then LCM := 0 else LCM := (a div GCD(a, b)) * b;
end;

function Solve(): Int64;
var
    i: Integer;
    ans: Int64;
begin
    ans := 1;
    for i := 2 to 20 do ans := LCM(ans, i);
end;

begin
    WriteLn('Smallest positive number evenly divisible by 1..20 is ', Solve);
end.

File: 5/5.c
#include <stdint.h>
#include <stdbool.h>
#include <stdio.h>

bool isDivisible(uint64_t dividend)
{
    static uint64_t divisors[10] = {
        11, 12, 13, 14, 15, 16, 17, 18, 19, 20,
    };

    for (uint64_t i = 0; i < 10; i++)
    {
        const uint64_t divisor = divisors[i];
        if (dividend % divisor != 0)
        {
            return false;
        }
    }
    return true;
}

uint64_t solve()
{
    uint64_t n = 1;
    while (!isDivisible(n))
    {
        n++;
    }
    return n;
}


int main()
{
    fprintf(stdout, "The smallest positive number that is evenly divisible by all of the numbers from 1 to 20 is %lu\n", solve());
    return 0;
}
